fossunited/fossunited/doctype/event_check_in/event_check_in.py

```diff
@@ -0,0 +1,22 @@
+# Copyright (c) 2024, Frappe x FOSSUnited and contributors
+# For license information, please see license.txt
+
+# import frappe
+from frappe.model.document import Document
+
+
+class EventCheckIn(Document):
+    # begin: auto-generated types
+    # This code is auto-generated. Do not modify anything in this block.
+
+    from typing import TYPE_CHECKING
+
+    if TYPE_CHECKING:
+        from frappe.types import DF
+
+        check_in_time: DF.Datetime | None
```

Comment:
@harshtandiya  it's not clear to me why we need an entirely new doctype instead of just adding `check_in_time` attribute to the existing `FOSSEventTicket` doctype.

Is the usecase is to enable multiple "check-ins" e.g. for multi-day events? Even in that case, `check_in_times` could be the attribute on the `FOSSEventTicket` doctype which is a list of datetimes or none and everytime check-in happens, the list will be appended to with the new check-in time.

A standalone doctype feels like overkill, maybe because I don't have sufficient context

Reply:
The usecase is indeed for multiple check in times for multi-day events.

For a field that is a list of datetimes, i am not sure if frappe would provide a fieldtype which will let us store a "list" of things. The thing that comes the closest is indeed a child doc/table which i did here.

Other than that there's JSON field type, which i feel will be messy in this use case.

Also, we can reuse this doctype if we decide to bring in check-ins for RSVPs too

Reply:
> For a field that is a list of datetimes, i am not sure if frappe would provide a fieldtype which will let us store a "list" of things. The thing that comes the closest is indeed a child doc/table which i did here.

okay then.

> Also, we can reuse this doctype if we decide to bring in check-ins for RSVPs too

The only reason we need to do check-ins is to provide meal coupons. Other than that, I don't see a strong reason why we should do them for conferences too. So I highly doubt we want to do it for meetup RSVPs. Also because the meetups will likely not have time to actually take "attendance"

Reply:
Makes sense